examples/ecs/generic_systems.rs

```diff
@@ -0,0 +1,27 @@
+use bevy::prelude::*;
+use bevy::ecs::component::Component;
+
+struct A;
+struct B;
+
+pub fn spawn_entities<T: Component>(mut commands: Commands,
+    mouse_button_inputs: Res<Input<MouseButton>>,) {
+    for _ in 0..10 {
+        let id = commands.spawn().insert(T).id();
+        if mouse_button_inputs.just_pressed(MouseButton::Left) {
+            return info!("Spawned entity {:?} with component {}", id, std::any::type_name::<T>());
+        }
+        if mouse_button_inputs.just_pressed(MouseButton::Right) {
+            return info!("Spawned entity {:?} with component {}", id, std::any::type_name::<T>());
+        }
```

Comment:
It's best to lift the `if` logic outside of the loop.
Otherwise were running `just_pressed` 10 times each system call. (This can be expensive since we have to hash the value, the loop it up in the set).

Reply:
So would something like this work better?

pub fn spawn_entities<T: Component>(mut commands: Commands,
    mouse_button_inputs: Res<Input<MouseButton>>,) {
    for _ in 0..10 {
        let id = commands.spawn().insert(T).id();
    }
        if mouse_button_inputs.just_pressed(MouseButton::Left) {
            return info!("Spawned entity {:?} with component {}", id, std::any::type_name::<T>());
        }
        if mouse_button_inputs.just_pressed(MouseButton::Right) {
            return info!("Spawned entity {:?} with component {}", id, std::any::type_name::<T>());
        }
}

Reply:
Essentially want to remove the for loop entirely or have the loop return "id"?

Reply:
The `id` part is necessary in the loop.
Rather I meant assigning the results of `just_pressed` outside of the loop and then referring to those instead of calling `just_pressed` multiple times.

Reply:
So, currently this is the function body:
```rust
for _ in 0..10 {
    let id = commands.spawn().insert(T).id();
    info!("Spawned entity {:?} with component {}", id, std::any::type_name::<T>()); 
}
mouse_button_inputs.just_pressed(MouseButton::Left);
mouse_button_inputs.just_pressed(MouseButton::Right);
```
Here you are throwing away the result of both `just_pressed` function calls. Meaning using them now is completely unnecessary. 

What I mean was to do something like:
```rust
let left_pressed = mouse_button_inputs.just_pressed(MouseButton::Left);
let right_pressed = mouse_button_inputs.just_pressed(MouseButton::Right);

for _ in 0..10 {
    if left_pressed { /* do something */ }
    if right_pressed { /* do something */ }
}
```
This way we save the values of `just_pressed` and then utilize them in our loops.

However, upon looking at it, the system is still quite confusing. 
I would recommend something like:

```rust
fn spawn_entities_on_click<T: Component, const N: usize>(
    mut cmds: Commands, 
    mouse_button_inputs: Res<Input<MouseButton>>,
) {
    if mouse_button_inputs.just_pressed(MouseButton::Left) {
        for _ in 0..N { 
            cmds.spawn().insert(T).id(); 
        }
        info!("spawned {} entities with component {}", N, std::any::type_name::<T>());
    }
}
```
This way we can also use `const generics` in our system which might be fun!

Reply:
@NathanSWard so for the Structs that I am using in the example could I change those or would
I still want to use them in the system?